File: leetcode/LinkedListTests/LinkedList_1290.cs
namespace LinkedListTests;

[TestFixture]
internal class LinkedList_1290
{
    private class Solution {
        public int GetDecimalValue(ListNode head) {
            var sum = 0;
            while(head is not null) {
                var data = head.val;
                sum = sum*2+data;
                head = head.next;
            }
            return sum;
        }
    }
}

File: leetcodePython/LinkedList/linkedlist_1290.py
from typing import Optional


class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def getDecimalValue(self, head: Optional[ListNode]) -> int:
        total = 0
        while head:
            total = total * 2 + head.val
            head = head.next
        return total
